zilliqa/tests/it/main.rs

```diff
@@ -672,14 +677,23 @@ impl Network {
             }
             AnyMessage::External(external_message) => {
                 let nodes: Vec<(usize, &TestNode)> = if let Some(destination) = destination {
-                    vec![self
+                    let (index, node) = self
                         .nodes
                         .iter()
                         .enumerate()
                         .find(|(_, n)| n.peer_id == destination)
-                        .unwrap()]
+                        .unwrap();
+                    if self.disconnected.contains(&index) {
```

Comment:
Maybe removing this `if` here (and the `filter` below) and then just adding `nodes.retain(|(i, _)| !self.disconnected.contains(i));` to cover both cases at once would be more concise - down to preference

Reply:
I still need a case to filter by the message's destination, if the node is not disconnected, but I think you're right that I can combine them.

I changed it to this, let me know if you think its worse :) 

```rust
                let nodes = self.nodes
                    .iter()
                    .enumerate()
                    .filter(|(i, n)| {
                        let disconnected = self.disconnected.contains(i);
                        // Send to this node if it is the intended destination or if the message is a broadcast.
                        let is_destination = destination.map(|d| n.peer_id == d).unwrap_or(true);

                        !disconnected && is_destination
                    });
```